push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/MessageWithVariants.java

```diff
@@ -0,0 +1,53 @@
+package org.jboss.aerogear.unifiedpush.message;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.jboss.aerogear.unifiedpush.api.PushMessageInformation;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.VariantType;
+
+public class MessageWithVariants implements Serializable {
```

Comment:
Similarly to `MessageWithTokens`:

It is a single-purpose serializable crate for specific JMS payload (didn't find any suitable existing type).

On later occasion we may also want to optimize how it gets serialized/deserialized.

Adding javadoc.

If you find a better name, I'm all for changing that!! :-)